diesel/src/connection/mod.rs

```diff
@@ -7,6 +8,8 @@ use result::*;
 use types::HasSqlType;
 
 pub use self::transaction_manager::{TransactionManager, AnsiTransactionManager};
+#[doc(hidden)]
+pub use self::statement_cache::StatementCacheKey;
```

Comment:
Probably, yes. (I actually don't mind third party connection adapters using this, I just don't want to commit to stability with its API)